File: LeetCode/1299-Replace-Elements-with-Greatest-Element-on-Right-Side.cs
public class Solution {
    public int[] ReplaceElements(int[] arr) {
        for (int i = 0; i < arr.Length; i++) 
        { 
            int max_value = -1; 
            for (int j = i + 1; j < arr.Length; j++) 
            { 
                max_value = Math.Max(max_value, arr[j]); 
            } 
            arr[i] = max_value; 
        }
        return arr;
    }
}

File: LeetCode/1299-Replace-Elements-with-Greatest-Element-on-Right-Side.js
/**
 * @param {number[]} arr
 * @return {number[]}
 */
var replaceElements = function(arr) {
    let max = arr[arr.length - 1];
    arr[arr.length - 1] = -1;
    
    for(let i = arr.length - 2; i >= 0; i--) {
        let curr = arr[i];
        arr[i] = max;
        if(curr > max) max = curr;
    }
    
    return arr;
};